cmd/cloudflared/config/configuration.go

```diff
@@ -63,6 +63,14 @@ func FindDefaultConfigPath() string {
 	return ""
 }
 
+// ValidateUnixSocket ensures --unix-socket param is used exclusively
+func ValidateUnixSocket(c *cli.Context) (string, error) {
+	if c.IsSet("unix-socket") && (c.IsSet("url") || c.NArg() > 0) {
+		return "", errors.New("--unix-socket must be used exclusivly.")
+	}
+	return c.String("unix-socket"), nil
+}
+
```

Comment:
This is a nice change: I like that ValidateUnixSocket is its own function now.

I'd appreciate it if you added a note on the ValidateUnixSocket documentation, saying that it must be mutually exclusive with ValidateUrl –– and an analogous note on ValidateUrl, saying that it must be mutually exclusive with ValidateUnixSocket. 

That way, if a programmer changed the ValidateUrl logic, they would remember to update the "url detection" part of ValidateUnixSocket.

Reply:
Maybe a little too expressive to my liking but here goes :)